### DISCUSSION TITLE
aedes concurrency value on cluster

### DISCUSSION THREAD

**Discussion by Gianluca-Casagrande-Stiga** [moscajs/aedes#770] [Q&A] [answered]
Hi,

on my mqtt broker I often have a "Possibile memory leak" error, due to TLSSocket listeners.
The maxListeners value now is set to 200, that is 2*concurrency option default value, as we can read on docs:

`concurrency <number> maximum number of concurrent messages delivered by mq. Default: 100`

I'm using the Redis mqemitter with a cluster configuration. My question is: the concurrency value should be related only to one broker, or I must consider the entire cluster messages throughput?
For example: if I had a 5 broker cluster that totally elaborates 1000 concurrency messages, I would set concurrency value to 1000 or 1000/5=200 ?

thanks

**Comment by robertsLando** [comment, marked as answer]
> For example: if I had a 5 broker cluster that totally elaborates 1000 concurrency messages, I would set concurrency value to 1000 or 1000/5=200 ?

1000, each broker must always reveive all messages in order to be able to send them to his connected clients. mqemitter is designed exactly for this, share messages across multiple instances
